iot/api/manager/src/main/java/com/google/cloud/iot/examples/DeviceRegistryExample.java

```diff
@@ -0,0 +1,338 @@
+ * IoT devices and registries, using both RSA and eliptic curve keys for authentication.
+ *
+ * <p>To start, follow the instructions on the Developer Guide at cloud.google.com/iot to create a
+ * service_account.json file. Also create a Cloud Pub/Sub topic as discussed in the guide. After
+ * doing this, you should have service_account.json in your directory. Tell gcloud about this file
+ * by setting it in an environment variable:
+ *
+ * <pre>
+ * <code>
+ * $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service_account.json
+ * </code>
+ * </pre>
+ *
+ * <p>Before running the example, we have to create private and public keys, as described in
+ * cloud.google.com/iot. Since we are interacting with the device manager, we will only use the
+ * public keys. The private keys are used to sign JWTs to authenticate devices. See the MQTT client
+ * example in the developer guide for an example of this.
+ *
+ * <p>Finally, compile and run the example with:
+ *
+ * <pre>
+ * <code>
+ * $ mvn clean compile assembly:single
+ * $ mvn exec:java \
+ *       -Dexec.mainClass="com.google.cloud.iot.examples.DeviceRegistryExample" \
+ *       -Dexec.args="-project_id=my-project-id \
+ *                    -pubsub_topic=projects/my-project-id/topics/my-topic-id \
+ *                    -ec_public_key_file=/path/to/ec_public.pem \
+ *                    -rsa_certificate_file=/path/to/rsa_cert.pem"
+ * </code>
+ * </pre>
+ */
+public class DeviceRegistryExample {
+  // CloudIot service for administering Cloud IoT devices, registries and projects.
+  private CloudIot service;
+  // Path to the project and location: "projects/my-project-id/locations/us-central1"
+  private String projectPath;
+  // Path to the registry: "projects/my-project-id/location/us-central1/registries/my-registry-id"
+  private String registryPath;
+
+  /** Construct a new registry with the given name and pubsub topic inside the given project. */
+  public DeviceRegistryExample(
+      String projectId, String location, String registryName, String pubsubTopicPath)
+      throws IOException, GeneralSecurityException {
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault().createScoped(CloudIotScopes.all());
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    service = new CloudIot(GoogleNetHttpTransport.newTrustedTransport(), jsonFactory, init);
+    projectPath = "projects/" + projectId + "/locations/" + location;
+    registryPath = projectPath + "/registries/" + registryName;
+
+    NotificationConfig notificationConfig = new NotificationConfig();
+    notificationConfig.setPubsubTopicName(pubsubTopicPath);
+
+    DeviceRegistry registry = new DeviceRegistry();
+    registry.setEventNotificationConfig(notificationConfig);
+    registry.setId(registryName);
+    service.projects().locations().registries().create(projectPath, registry).execute();
```

Comment:
Ack, maybe better as a builder, I'm not sure how much control we have over this as it's API client.